File: rust/stack.rs

pub struct Stack<T> {
    elements: Vec<T>,
}

impl <T> Stack<T> {

    pub fn new() -> Self {
        Stack { elements: Vec::new() }
    }


    pub fn push(&mut self, item:T) {
        self.elements.push(item);
    }


    pub fn pop(&mut self) -> Option<T> {
        self.elements.pop()
    }

    pub fn peek(&self) -> Option<&T>{
        self.elements.last()
    }

    pub fn size(&self) -> usize {
        self.elements.len()
    }

    pub fn is_empty(&self) -> bool {
        self.elements.is_empty()
    }

}


fn main() {
    // Create a new stack
    let mut stack: Stack<i32> = Stack::new();

    // Push elements onto the stack
    stack.push(1);
    stack.push(2);
    stack.push(3);

    // Peek at the top element
    println!("Top element is: {:?}", stack.peek());

    // Pop elements from the stack
    println!("Popped element is: {:?}", stack.pop());
    println!("Popped element is: {:?}", stack.pop());

    // Check if the stack is empty
    println!("Is the stack empty? {}", stack.is_empty());

    // Check the size of the stack
    println!("Stack size is: {}", stack.size());
}

File: python/stack.py
class Stack :
    def __init__(self) -> None:
        self.elements = []

    def push(self, element) -> None:
        self.elements.append(element)

    def pop(self):
        if(self.is_empty()):
            raise Exception("Your Stack is empty cannot pop the elements")
        return self.elements.pop()
    
    def peek(self):
        if(self.is_empty()):
            raise Exception("Your Stack is empty so there is no element to peek at")
        return self.elements[-1]

    def is_empty(self):
        if((len(self.elements)) == 0):
            return True
        return False
    
    def size(self):
        return len(self.elements)


def main():
    st = Stack()

    print(st.is_empty())
    print("Oh! yes it is empty")
    print(st.push(2))
    print("Lol None print hoga")
    st.push(1)
    st.push(3)

    print(st.peek())
    print(st.pop())
    print(st.peek())

    print(st.size())

if  __name__  == "__main__":
    main()

